### PR TITLE
feat: hashlogger wiring cleanup

### LINKED ISSUE #13401: Clean up hash logger wiring
Convert the wiring for the hash logger into the more modern format.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13401

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 1]
```diff
@@ -1035,12 +1038,13 @@ public StateHasher buildStateHasher() {
         return stateHasher;
     }
 
-    /**
+    /*
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1041]: formatting got messed up here

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java [hunk 2]
```diff
@@ -112,8 +111,8 @@ public record PlatformSchedulersConfig(
                 TaskSchedulerConfiguration runningEventHasher,
         @ConfigProperty(defaultValue = "SEQUENTIAL CAPACITY(500) UNHANDLED_TASK_METRIC")
                 TaskSchedulerConfiguration issDetector,
-        @ConfigProperty(defaultValue = "SEQUENTIAL_THREAD") TaskSchedulerType hashLoggerSchedulerType,
-        @ConfigProperty(defaultValue = "100") int hashLoggerUnhandledTaskCapacity,
+        @ConfigProperty(defaultValue = "SEQUENTIAL CAPACITY(100) UNHANDLED_TASK_METRIC")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:114]: Was the change from `SEQUENTIAL_THREAD` to `SEQUENTIAL` intentional?

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulersConfig.java:114]: Intentional. This thread is just logging, there is no need to have a dedicated thread. Although technically it touches the state, it only touches in memory parts (so we don't have to worry about disk access).

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1041]: fixed formatting

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 3]
```diff
@@ -1071,4 +1074,35 @@ public StateSnapshotManager buildStateSnapshotManager() {
...
+    /**
+     * Provide a hash logger in place of the platform's default hash logger.
+     *
+     * @param hashLogger the hash logger to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withHashLogger(@NonNull final HashLogger hashLogger) {
+        throwIfAlreadyUsed();
+        if (this.hashLogger != null) {
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:1087]: this is a bug as an exception is thrown whether the hashlogger is null or not.

**Review by kfa-aguda — APPROVED** [review]

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**